common/src/main/java/com/twoez/common/HTMLReaderHelper.java

```diff
@@ -0,0 +1,137 @@
+package com.twoez.common;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayDeque;
+import java.util.Queue;
+
+public class HTMLReaderHelper {
+
+    private final URL currentURL;
+
+    public HTMLReaderHelper(URL url){
+        currentURL = url;
+    }
+
+    public CharIterator charIterator(){
+        CharIterator iterator;
+        try {
+            iterator = new CharIterator();
+        } catch(IOException ex){
+            return null;
+        }
+        return iterator;
+    }
+
+    public StringIterator stringIterator(){
+        return new StringIterator();
+    }
+
+    public class CharIterator implements java.util.Iterator<Character>{
```

Comment:
You really need to hate libraries in order to come up with this :) 
What was wrong with BufferedReader ?

Reply:
We thought that buffering policy in BufferedReader might slow down getting price. So we rewrote it. But that was reasonable until we found out that `mark()` and `reset()` methods are not supported by our JDK

Reply:
BufferedReader should be fine in your case.
One scenario where it would be ineffective is when the whole HTML is formatted into single line, but it is not the case with xmarkets. Premature optimization is evil